eng/pipelines/templates/stages/archetype-js-release.yml

```diff
@@ -9,12 +9,14 @@ parameters:
   TargetDocRepoName: ''
   
 stages:
-  - ${{if and(eq(variables['Build.Reason'], 'Manual'), eq(variables['System.TeamProject'], 'internal'))}}:
+  - ${{if and(or(eq(variables['TestPipeline'], 'true'), eq(variables['Build.Reason'], 'Manual')), eq(variables['System.TeamProject'], 'internal'))}}:
     - ${{ each artifact in parameters.Artifacts }}:
       - stage: Release_${{artifact.safeName}}
         displayName: 'Release: ${{artifact.name}}'
         dependsOn: ${{parameters.DependsOn}}
-        condition: and(succeeded(), ne(variables['SetDevVersion'], 'true'), ne(variables['Skip.Release'], 'true'), ne(variables['Build.Repository.Name'], 'Azure/azure-sdk-for-js-pr'))
+        condition: >-
+          and(succeeded(), or(ne(variables['SetDevVersion'], 'true'), eq(variables['TestPipeline'], 'true')),
```

Comment:
Do we need this as part of this condition as well?

Reply:
Seems to me like we do. What do you think?

Reply:
Won't this cause the changes in eng/common to trigger actual release of packages?

Reply:
Yes we are planning to allow this to run automatically up to the approval gate. it will still require the approval before the actual release can happen.

Reply:
@chidozieononiwu Do you plan to set SetDevVersion=true as well? If so then yes I guess you do need to add the TestPipeline condition here, if you aren't then I wouldn't expect you to need this condition.

Reply:
Ok, I see what you mean, the variable does not necessarily need to be set for it to be a dev version.